src/toil/jobStores/aws/jobStore.py

```diff
@@ -1504,27 +1496,23 @@ def __repr__(self):
 
     versionings = dict(Enabled=True, Disabled=False, Suspended=None)
 
-    def _getBucketVersioning(self, bucket):
+    def _getBucketVersioning(self, bucket_name):
         """
-        For newly created buckets get_versioning_status returns an empty dict. In the past we've
-        seen None in this case. We map both to a return value of False.
-
-        Otherwise, the 'Versioning' entry in the dictionary returned by get_versioning_status can
-        be 'Enabled', 'Suspended' or 'Disabled' which we map to True, None and False
-        respectively. Note that we've never seen a versioning status of 'Disabled', only the
-        empty dictionary. Calling configure_versioning with False on a bucket will cause
-        get_versioning_status to then return 'Suspended' even on a new bucket that never had
-        versioning enabled.
+        The status attribute of BucketVersioning can be 'Enabled', 'Suspended' or None (Disabled)
+        which we map to True, None and False respectively. Note that we've never seen a versioning
+        status of 'Disabled', only the None return value. Calling BucketVersioning.suspend() will
+        cause BucketVersioning.status to then return 'Suspended' even on a new bucket that never
+        had versioning enabled.
+
+        :param bucket_name: str
         """
-        for attempt in retry_s3():
-            with attempt:
-                status = bucket.get_versioning_status()
-                return self.versionings[status['Versioning']] if status else False
+        status = self.s3_resource.BucketVersioning(bucket_name).status
+        return self.versionings[status] if status else False
 
-    def __getBucketRegion(self, bucket):
-        for attempt in retry_s3():
-            with attempt:
-                return bucket_location_to_region(bucket.get_location())
+    @staticmethod
+    def __getBucketRegion(bucket_name):
```

Comment:
Potentially rename this to just `def getBucketRegion(bucket_name)` without the `__` as it's not appropriate here imo.